packages/react-router/docs/api/matchPath.md

```diff
@@ -29,3 +29,34 @@ to the matching props `Route` accepts:
     exact; // optional, defaults to false
 }
 ```
+
+## returns
+
+It returns `null` when provided pathname does not match `path` prop or an object when it does. 
```

Comment:
That sentence reads a bit awkwards. I suggest inverting it. That would also match the order of your examples below:

> It returns an object if the provided pathname matches the `path` prop or `null` if it does not.